test/integration/integration_admin_test.cc

```diff
@@ -442,4 +444,78 @@ TEST_F(IntegrationAdminIpv4Ipv6Test, Ipv4Ipv6Listen) {
   }
 }
 
+// Testing the behavior of StatsMatcher, which allows/denies the  instantiation of stats based on
+// restrictions on their names.
+class StatsMatcherIntegrationTest : public HttpIntegrationTest, public testing::Test {
+public:
+  StatsMatcherIntegrationTest()
+      : HttpIntegrationTest(Http::CodecClient::Type::HTTP1, Network::Address::IpVersion::v4,
+                            simTime()) {}
+
+  void initialize() override {
+    config_helper_.addConfigModifier(
+        [this](envoy::config::bootstrap::v2::Bootstrap& bootstrap) -> void {
+          *bootstrap.mutable_stats_config()->mutable_stats_matcher() = stats_matcher_;
+        });
+    HttpIntegrationTest::initialize();
+
+    response_ = IntegrationUtil::makeSingleRequest(lookupPort("admin"), "GET", "/stats", "",
+                                                   downstreamProtocol(), version_);
+  }
+
+  void TearDown() override {
+    EXPECT_TRUE(response_->complete());
+    EXPECT_STREQ("200", response_->headers().Status()->value().c_str());
+
+    test_server_.reset();
+    fake_upstreams_.clear();
```

Comment:
You probably don't need these; gtest will clean these out at the end of each test when it reconstructs the Test class instance.

Reply:
Done.